code/game/machinery/embedded_controller/airlock_controllers.dm

```diff
@@ -210,4 +395,51 @@
 	if(clean)
 		program.receive_user_command(href_list["command"])
 
-	return 1
\ No newline at end of file
+	return 1
+
+/obj/machinery/embedded_controller/radio/access_controller/linkMenu(var/obj/O)
+	var/dat = ""
+	if(istype(O, /obj/machinery/door/airlock))
+		dat += "<a href='?src=\ref[src];link=1;slot=ext'>\[Link @ exterior\]</a><br>"
+		dat += "<a href='?src=\ref[src];link=1;slot=int'>\[Link @ interior\]</a>"
+	return dat
+
+/obj/machinery/embedded_controller/radio/access_controller/canLink(var/obj/O, var/list/context)
+	if(istype(O, /obj/machinery/door))
+		if(context["slot"] in list("int", "ext"))
+			return 1
+
+	return 0
+
+/obj/machinery/embedded_controller/radio/access_controller/unlinkFrom(var/mob/user, var/obj/O)
+	if(!("id_tag" in O.vars))//erm, stop HTML Injecting you! (can you even HTML inject this?) ((this can't happen, atleast not on this UI without cheating)), and tbh why would you even do this.
+		return 0
+
+	if(istype(O, /obj/machinery/door))
+		if(tag_exterior_door == O:id_tag)
+			tag_exterior_door = null
+			return 1
+		else if(tag_interior_door == O:id_tag)
+			tag_interior_door = null
+			return 1
+
+/obj/machinery/embedded_controller/radio/access_controller/isLinkedWith(var/obj/O)
+	if(!("id_tag" in O.vars))//erm, stop HTML Injecting you! (can you even HTML inject this?) ((this can't happen, atleast not on this UI without cheating)), and tbh why would you even do this.
+		return 0
+
+	if(istype(O, /obj/machinery/door))
+		if(tag_interior_door  == O:id_tag)
```

Comment:
Wait, am I supposed to cast for every type? (Aurlock sensor, airlock, vent) because that seems sorta inneficient, I only did it with the access controller becuse that's the only one where tjere's only 1 typz